File: C#.Net/JDI/Web/Selenium/Elements/Common/FileInput.cs
using System;
using Epam.JDI.Core;
using Epam.JDI.Core.Interfaces.Common;
using Epam.JDI.Web.Selenium.Base;
using OpenQA.Selenium;

namespace Epam.JDI.Web.Selenium.Elements.Common
{
    public class FileInput : TextField, IFileInput
    {
        public FileInput() : this (null) { }
        public FileInput(By byLocator = null, IWebElement webElement = null)
            : base(byLocator, webElement) { }
        
        protected new Action<WebBaseElement, string> SetValueAction = (el, val) => ((FileInput) el).Input(val);
    }
}

File: Python/JDI/web/selenium/elements/common/file_input.py
from JDI.web.selenium.elements.common.text_field import TextField


class FileInput(TextField):

    def __init__(self, by_locator=None, by_label=None, web_element=None):
        super(FileInput, self).__init__(by_locator)

    def set_value_action(self, value):
        self.input(value)